### PR TITLE
support multilanguage on header, home

### PR DESCRIPTION
**Pull request by Moon-DaeSeung**
// 한국어
image

// english
image

### CODE REVIEW

File: constants/constants.ts [hunk 1]
```diff
@@ -1,30 +1,3 @@
...
-    title: "Blog",
-    href: "https://nextjs-blog-template.pages.dev/",
-  },
-  {
-    title: "Github",
-    href: "https://github.com/notionpresso",
-  },
-] as const;
-
 export const LANGUAGELIST = [
```
Comment (anonymousRecords) [inline constants/constants.ts:1]: p3: I think it will not be needed anymore.

**Review by anonymousRecords — APPROVED** [review]

File: app/[lang]/contributing/page.tsx [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 
-export async function generateStaticParams() {
-  return [{ lang: "ko" }, { lang: "en" }];
-}
+export const generateStaticParams = () => {
+  return SUPPORTED_LANGUAGES.map((lang) => ({ lang }));
+};
 
 export default function TutorialPage({ params }: { params: { lang: string } }) {
   const content = params.lang === "ko" ? kr : en;
```
[the remark this answers is not in the document]
Comment (Moon-DaeSeung) [reply app/[lang]/contributing/page.tsx]: good I fixed it

and TODO: dynamic import .json depending on lang

File: app/[lang]/page.tsx [hunk 3]
```diff
@@ -1,7 +1,12 @@
 import { LandingHero } from "@/components/landing/hero";
+import { SUPPORTED_LANGUAGES } from "@/i18n/supported-languages";
 
 export const runtime = "edge";
 
+export const generateStaticParams = () => {
+  return SUPPORTED_LANGUAGES.map((lang) => ({ lang }));
+};
+
```
[the remark this answers is not in the document]
Comment (Moon-DaeSeung) [reply app/[lang]/page.tsx]: I split this function on '@/i18n'

File: components/ui/header.tsx [hunk 4]
```diff
@@ -30,19 +34,46 @@ export default function Header() {
     return ["Blog", "Github"].includes(title);
   };
 
+  const items = [
```
[the remark this answers is not in the document]
Comment (Moon-DaeSeung) [reply components/ui/header.tsx:37]: as you can see, HEADER_LIST is not constant anymore (depending on lang)

and HEADER_LIST is only used on header component. so you'd not better split code into other files

File: constants/constants.ts (hunk 1 above)
Comment (Moon-DaeSeung) [reply constants/constants.ts]: do you mean LANGUAGE_LIST?

**Pull request merged**